packages/nextjs-demo/src/app/recipe/page.tsx

```diff
@@ -73,33 +74,36 @@ export default function RecipeWrapper({ searchParams }: { searchParams: any }) {
 
       <ResultContainer title={`AI comes up with a recipe for ${query}`}>
-        <AIComponent renderPassedReactComponents>
-          <ChatCompletion temperature={1}>
-            <SystemMessage>
-              You are an AI who is an expert chef and also an expert UI designer. The user will ask you for a recipe.
-              Your response must be structured using a set of React components. Here are the React components, and an
-              example of how they should be used:
-              <Recipe>the entire recipe contents</Recipe>
-              <RecipeTitle>the title of your recipe</RecipeTitle>
-              <RecipeInstructionList>
-                <RecipeInstructionListItem>the first instruction</RecipeInstructionListItem>
-              </RecipeInstructionList>
-              <RecipeIngredientList>
-                <RecipeIngredientListItem>the first ingredient</RecipeIngredientListItem>
-              </RecipeIngredientList>
-              Every child of a RecipeInstructionList should be a RecipeInstructionListItem. Every child of a
-              RecipeIngredientList should be a RecipeIngredientListItem. Respond with a JSON object that encodes your
-              UI. The JSON object should match this TypeScript interface: interface Element {'{'}
-              name: string; children: (string | Element)[]
-              {'}'}
-              For example:
-              {'{'}
-              "name": "Recipe", "children": [{'{'}"name": "RecipeTitle", "children": ["My Recipe"]{'}'}
-              "my description" ]{'}'}. Respond with only the JSON. Do not include with an explanatory suffix or prefix.
-            </SystemMessage>
-            <UserMessage>Give me a recipe for {query}. Respond with only the JSON.</UserMessage>
-          </ChatCompletion>
-        </AIComponent>
+        <Suspense fallback={'Loading...'}>
+          <AI.jsx>
+            <UICompletion
+              example={
+                <Recipe>
+                  <RecipeTitle>Crème Chantilly</RecipeTitle>
+                  <RecipeIngredientList>
+                    <RecipeIngredientListItem>2 cups heavy cream</RecipeIngredientListItem>
+                    <RecipeIngredientListItem>2 tablespoons granulated sugar</RecipeIngredientListItem>
+                    <RecipeIngredientListItem>1 teaspoon vanilla extract</RecipeIngredientListItem>
+                  </RecipeIngredientList>
+                  <RecipeInstructionList>
+                    <RecipeInstructionListItem>
+                      Combine the ingredients in a large mixing bowl.
+                    </RecipeInstructionListItem>
+                    <RecipeInstructionListItem>
+                      Beat the contents on high speed until soft peaks form.
+                    </RecipeInstructionListItem>
+                    <RecipeIngredientListItem>Keep chilled until serving.</RecipeIngredientListItem>
+                  </RecipeInstructionList>
+                </Recipe>
```

Comment:
The fact that we can still pass pure React as a prop make me feel much better about the "embedding components is not supported" caveat.

Reply:
It actually is supported now! And I'm using it in the "render-to-html" demo.